### DISCUSSION TITLE
Clink completion with set /p (batch windows)

### DISCUSSION THREAD

**Discussion by boka-boka** [chrisant996/clink#552] [Q&A] [answered]
Hello,

I have a very simple bat file that works well in standard cmd, but doesn't work when executed in a clink context:

```batchfile
@echo off
set /p test=enter file name using tab completion:
call :PRINT
goto :EOF

:PRINT
echo test is : "%test%"
goto :EOF
``` 

Idea is to retrieve the name of a file found through completion capability. When I launch script, and use "tab" to complete to any file in the current directory, output is different with clink compared to basic cmd.
With cmd:
```batchfile
C:\tmp>c:\tmp\test.bat
enter file name using tab completion:myfile.txt
test is : "myfile.txt"
```

With clink/cmd:
```batchfile
C:\tmp>c:\tmp\test.bat
enter file name using tab completion:myfile.txt
test is : ""
```

Should I proceed differently ? Is that a known behavior ?
Thank you,
Erwan

**Comment by chrisant996** [comment]
@boka-boka Thanks for reporting this.

> When I launch script, and use "tab" to complete to any file in the current directory, output is different with clink compared to basic cmd.

When I run the script in a cmd with clink injected, <kbd>Tab</kbd> doesn't even do completion; it just inserts raw TAB characters.

Once I debug and fix whatever is causing <kbd>Tab</kbd> to not work, then I'll be able to debug and fix the problem where the input isn't captured properly.

P.S.  This used to work as of v1.4.0, so something has regressed since then.

> **v1.4.0**
> - Fixed [#370](https://github.com/chrisant996/clink/issues/370); <kbd>Tab</kbd> inside `set /p var=` mistakenly discards any previous input.

**Reply by chrisant996** [reply]
Opened issue #553 to track the problem and resolution.

**Reply by chrisant996** [reply, marked as answer]
The reason completion didn't work for me was for some reason I configured my CMD windows to disable plain CMD completion.

With completion enabled, then the script works as expected for me.

Can we continue the investigation in issue #553?  I have follow-up questions there.
